searchguard/rolesmapping.py

```diff
@@ -152,3 +165,26 @@ def modify_rolemapping(role, properties, action="replace"):
 
     # No merge or split action, overwrite existing properties:
     _send_api_request(role, properties)
+
+
+def list_rolemappings_for_user(user, roles=None):
+    """Get list of rolemappings that contain the given user. If no roles are provided,
+    all rolemappings are checked.
+
+    :param str user: Name of user
+    :param list roles: List of rolemappings to be checked for the given user
+    :returns list: list of rolemappings with the given user
+    :raises: ViewRoleMappingException
+    """
+    user_rolemappings = []
+
+    if roles:
+        for role in roles:
```

Comment:
you could do something like this instead here:
```
user_role_mappings = [role for role in roles if user in view_rolemapping(role)[role]['users']]
```
also instead of appending the user_role_mappings in an if and else you can just define it directly there as there is nothing else adding or removing stuff from that list